charts/karpenter/templates/clusterrole.yaml

```diff
@@ -37,5 +37,27 @@ rules:
     resources: ["daemonsets"]
     verbs: ["list", "watch"]
   - apiGroups: ["admissionregistration.k8s.io"]
-    resources: ["validatingwebhookconfigurations", "mutatingwebhookconfigurations"]
-    verbs: ["get", "watch", "list", "update"]
+    resources: ["mutatingwebhookconfigurations", "validatingwebhookconfigurations"]
+    verbs: ["list", "watch"]
+  - apiGroups: ["admissionregistration.k8s.io"]
+    resources: ["mutatingwebhookconfigurations"]
+    verbs: ["get", "update", "delete"]
+    resourceNames: ["defaulting.webhook.provisioners.karpenter.sh"]
+  - apiGroups: ["admissionregistration.k8s.io"]
+    resources: ["validatingwebhookconfigurations"]
+    verbs: ["get", "update", "delete"]
+    resourceNames: ["validation.webhook.provisioners.karpenter.sh", "validation.webhook.config.karpenter.sh"]
+  - apiGroups: [""]
+    resources: ["namespaces/finalizers"]
+    verbs: ["update"]
+    resourceNames:
+      - "{{ .Release.Namespace }}"
+  - apiGroups: [""]
+    resources: ["replicationcontrollers/finalizers"]
+    verbs: ["update"]
+  - apiGroups: ["apps"]
+    resources: ["statefulsets/finalizers", "replicasets/finalizers"]
+    verbs: ["update"]
+  - apiGroups: ["batch"]
+    resources: ["jobs/finalizers"]
+    verbs: ["update"]
```

Comment:
If the permissions are not granted and the admission controller [OwnerReferencesPermissionEnforcement](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#ownerreferencespermissionenforcement) is enabled, errors are thown like 

> 2022-04-09T20:43:33.612Z	ERROR	controller.provisioning	Failed to bind default/nginx-6cd649d446-qx4q6 to i-09b156a361a1b17ac.eu-central-1.compute.internal, pods "nginx-6cd649d446-qx4q6" is forbidden: cannot set blockOwnerDeletion if an ownerReference refers to a resource you can't set finalizers on: , <nil>	{"commit": "8280611", "provisioner": "default"}

If the ownerReference of a pod is a replicasets, karpenter needs permissions to set finalizers on the ownerReference.

Reply:
But karpenter doesn't need to update finalizers on pods.

Reply:
> But karpenter doesn't need to update finalizers on pods.

Thats true. But this PR also not granted finalizers permissions on pods.

Reply:
I'm not sure I understand. If we're not adding finalizers to pods, why do we need permission to finalizers on all these resource types?

Reply:
I have no clue whats going inside karpeter. If the permissions are omit, permissions errors are raised on the controller logs:

> 2022-04-09T20:43:33.612Z ERROR controller.provisioning Failed to bind default/nginx-6cd649d446-qx4q6 to i-09b156a361a1b17ac.eu-central-1.compute.internal, pods "nginx-6cd649d446-qx4q6" is forbidden: cannot set blockOwnerDeletion if an ownerReference refers to a resource you can't set finalizers on: , {"commit": "8280611", "provisioner": "default"}

I guess, since https://github.com/aws/karpenter/pull/1856 is merged, the RBAC permissions can be removed, too. But I'm  currently not able to validate this. 

please consider to enable [OwnerReferencesPermissionEnforcement](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#ownerreferencespermissionenforcement) admission controller on the kuberneteres cluster inside your CI.